core/src/main/java/uk/ac/bbsrc/tgac/miso/core/service/naming/generation/OicrSampleAliasGenerator.java

```diff
@@ -43,7 +43,12 @@ public String generate(Sample sample) throws MisoNamingException, IOException {
         return addSiblingTag(parent.getAlias(), detailed);
       }
       if (isTissueSample(parent)) {
-        return addSiblingTag(parent.getAlias(), detailed);
+        if (isTissueSample(detailed) && isIdentitySample(parent.getParent())) {
```

Comment:
Done, and also removed the check for if the identity is the immediate grandparent that was immediately above the `// tissues parented to tissues` line